common/rpc/interceptor/namespace_validator.go

```diff
@@ -192,18 +195,26 @@ func (ni *NamespaceValidatorInterceptor) StateValidationIntercept(
 		return nil, err
 	}
 
-	err = ni.checkNamespaceState(namespaceEntry, info.FullMethod)
-	if err != nil {
-		return nil, err
-	}
-	err = ni.checkReplicationState(namespaceEntry, info.FullMethod)
-	if err != nil {
+	if err := ni.ValidateState(namespaceEntry, info.FullMethod); err != nil {
 		return nil, err
 	}
 
 	return handler(ctx, req)
 }
 
+// ValidateState validates:
+// 1. Namespace is specified in task token if there is a `task_token` field.
+// 2. Namespace is specified in request if there is a `namespace` field and no `task_token` field.
+// 3. Namespace exists.
+// 4. Namespace from request match namespace from task token, if check is enabled with dynamic config.
+// 5. Namespace is in correct state.
+func (ni *NamespaceValidatorInterceptor) ValidateState(namespaceEntry *namespace.Namespace, fullMethod string) error {
```

Comment:
Should we remove `Interceptor` from the type name since it is redundant with the package name?

Reply:
No, this package exports multiple interceptors and other things as well, I think it's better to clarify that this is an interceptor in the name.